xdsl/parser.py

```diff
@@ -724,6 +725,10 @@ def parse_integer_type():
         if int_type := self.try_parse(parse_integer_type):
             return int_type
 
+        # Shorthand for DenseArrayBase
+        if (dense := self.parse_optional_mlir_dense_array_base()) is not None:
```

Comment:
I dont think you need the `is not None` part, right?

Reply:
`dense` might not be None but be another falsey value, the `is not None` is better. The type checker prefers it also in most cases

Reply:
So this is something that I know @georgebisbas has different opinion than me.
I think that `is not None` is more explicit, and way less error prone, since we won't break it because of a refactor for instance. Though yes, it is redundant.

Since we already uniformized the code by removing the `is not None` in the other files, I'll remove it here as well to be coherent!

Reply:
I would argue that we should always have `not None`, since this is more robust in general, and convey the right information.
We want to check that something is not `None`, not that something is null, if that makes sense. Even if `is null` == `is None` in these cases, if we change our API this will break without any warning/errors.